api/src/models/user/user.methods.ts

```diff
@@ -194,10 +194,15 @@ export async function createAndPostThread(this: IUserDocument, threadDetails: IT
     likes: {},
     shares: {}
   };
+  try {
+    const newlyCreatedThread = await ThreadModel.create(userThread);
+    this.threads.started[`${newlyCreatedThread.id.toString()}`] = newlyCreatedThread;
+    await this.save();
+    return [this, newlyCreatedThread];
```

Comment:
Maybe this would self describe it's self better as an object... something like {userData: this, threadData: newlyCreatedThread}